app/views/purchases/_form.html.erb

```diff
@@ -0,0 +1,33 @@
+<%= form_with(model: [(@category || Category.find(params[:category_id])), @purchase], local: true, class: 'purchase-form') do |form| %>
+  <% if @purchase.errors.any? %>
+    <div class="errors">
+      <h2><%= pluralize(@purchase.errors.count, "error") %> prohibited this purchase from being saved:</h2>
+      <ul>
+        <% @purchase.errors.each do |error| %>
+          <li><%= error.full_message %></li>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
+
+  <div class="form-fields">
+  <div class="field">
+    <%= form.label :name %>
+    <%= form.text_field :name %>
+  </div>
+
+  <div class="field">
+    <%= form.label :amount %>
+    <%= form.text_field :amount %>
+  </div>
```

Comment:
- Well validated
- [ ] [OPTIONAL] It will be more professional to display some alert message to users as to why their new purchase is not been granted, For example: For the amount field "Kindly input n amount less that $1,000"